src/engine/tools/engine-suite/src/engine_test/integration.py

```diff
@@ -42,66 +54,82 @@ def __init__(self, args):
             exit(1)
 
         self.args['full_location'] = self.format.get_full_location(self.args)
-
+        # TODO: move escape :| :
         # Client to API TEST
         self.api_client = ApiConnector(args)
         self.api_client.create_session()
 
-    def run(self, interactive: bool = True):
-        loop = True
-        event_passed = self.args['event'] if 'event' in self.args else None
-        events = []
+    def run(self):
         events_parsed = []
         try:
-            while (loop):
-                loop = interactive
-
+            while True:
                 try:
+                    events = []
                     # Get the events
-                    events = EventsCollector.collect(interactive, self.format, event_passed)
+                    events = EventsCollector.collect(self.format)
+                    # Split the events
+                    events = self.format.get_events(events)
+                    # Format each event
+                    events = [self.format.format_event(event) for event in events]
+                    # Remove invalid events
+                    events = list(filter(None, events))
+
                     if len(events) > 0:
                         for event in events:
                             response = self.process_event(event, self.format)
                             events_parsed.append(response)
                 except KeyboardInterrupt as ex:
-                    loop = False
+                    break;
+
+                if not sys.stdin.isatty():
+                    break;
 
         except Exception as ex:
-            print("An error occurred while trying to process the events. Error: {}".format(ex))
+            print("An error occurred while trying to process the events of the integration. Error: {}".format(ex))
+            ex.with_traceback()
+
         finally:
             self.write_output_file(events_parsed)
             self.api_client.delete_session()
 
     def process_event(self, event, format):
-        event = format.format_event(event)
-        result = self.api_client.test_run(event)
-        response = "\n"
-        response_output = { }
-        response_traces = { }
-
-        if len(result["data"]["run"]["traces"]) > 0:
-            response_traces['Traces'] = result["data"]["run"]["traces"]
-
-        response_output['Output'] = result["data"]["run"]["output"]
 
-        if not self.args['json_format']:
-            if len(result["data"]["run"]["traces"]) > 0:
-                traces = self.response_to_yml(response_traces)
-                response += traces.replace("Traces", "---\nTraces")
+        # Get the values to send
+        result = self.api_client.test_run(event)
```

Comment:
question - desing: should use protobuf here?

Reply:
Additional work is currently needed for protobuf, it needs to be applied to the entire package, so it is pending for now.